src/vision/status_quo/barbara_needs_async_helpers.md

```diff
@@ -0,0 +1,64 @@
+# 😱 Status quo stories: Barbara needs Async Helpers
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+[Barbara], an experienced Rust user, is prototyping an async Rust service for work. To get things working quickly, she decides to prototype in tokio, since it is unclear which runtime her work will use.
+
+She starts adding warp and tokio to her dependencies list. She notices that warp suggests using tokio with the `full` feature. She's a bit concerned about how this might affect the compile times and also that *all* of tokio is needed for her little project, but she pushes forward.
+
+As she builds out functionality, she's pleased to see tokio provides a bunch of helpers like `join!` and async versions of the standard library types like channels and mutexes.
+
+After completing one endpoint, she moves to a new one which requires streaming http responses to the client. Barbara quickly finds out that tokio does not provide a stream type, and so she adds `tokio-stream` to her dependencies.
+
+Moving on she tries to make some functions generic over the web framework underneath, so she tries to abstract off the functionality to a trait. But Rust doesn't support async functions in traits yet, so she adds `async_trait` to her dependencies.
```

Comment:
Something like this:

---

She writes an async function inside a trait, just like a normal function.

```rust=
trait Example {
    async fn foo();
}
```

Then she gets a helpful error message.

```=
error[E0706]: functions in traits cannot be declared `async`
 --> src/lib.rs:2:5
  |
2 |     async fn foo();
  |     -----^^^^^^^^^^
  |     |
  |     `async` because of this
  |
  = note: `async` trait functions are not currently supported
  = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
```

---

I'll add this, and also the link to the crate.

Reply:
thanks :)